docs/notebooks/active_learning.pct.py

```diff
@@ -0,0 +1,177 @@
+    return tf.math.log(branin(x))
+
+
+search_space = Box([0, 0], [1, 1])
+
+fig = plot_function_plotly(log_branin, search_space.lower, search_space.upper, grid_density=20)
+fig.update_layout(height=400, width=400)
+fig.show()
+
+# %% [markdown]
+# We begin our Bayesian active learning steps after evaluating two initial points sampled by Halton sequence.
+
+# %%
+import trieste
+
+observer = trieste.utils.objectives.mk_observer(log_branin)
+
+num_initial_points = 2
+initial_query_points = search_space.sample_halton(num_initial_points)
+initial_data = observer(initial_query_points)
+
+
+# %% [markdown]
+# ## Surrogate model
+#
+# Just like in sequential optimization, we fit a surrogate Gaussian process model to the initial data.
+
+# %%
+import gpflow
+
+
+def build_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    kernel = gpflow.kernels.RBF(variance=variance, lengthscales=[0.2, 0.2])
+    gpr = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+    gpflow.set_trainable(gpr.likelihood, False)
+
+    return {
+        "model": gpr,
+        "optimizer": gpflow.optimizers.Scipy(),
+        "optimizer_args": {
+            "minimize_args": {"options": dict(maxiter=100)},
+        },
+    }
+
+
+model = build_model(initial_data)
+
+# %% [markdown]
+# ## Active learning using predictive variance
+#
+# To do our first active learning example, we will use `PredictiveVariance`, one of the simplest active learning acquisition function. the `PredictiveVariance` sample a point on highest determinant of the predictive covariance of our model. Here, we will consider single query point using `PredictiveVariance`, the batch case will be shown later. We can utilize trieste's `BayesianOptimizer` to do the active learning steps.
+
+# %%
+from trieste.acquisition.optimizer import generate_continuous_optimizer
+from trieste.acquisition.rule import EfficientGlobalOptimization
+from trieste.acquisition.function import PredictiveVariance
+
+acq = PredictiveVariance()
+rule = EfficientGlobalOptimization(
+    builder=acq, optimizer=generate_continuous_optimizer(sigmoid=False)
+)
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+
+# %% [markdown]
+# To plot the contour of variance of our model at each step, we can set the `track_state` parameter to `True` in `bo.optimize()`, this will make trieste record our model at each iteration.
+
+# %%
+bo_iter = 5
+result = bo.optimize(bo_iter, initial_data, model, rule, track_state=True)
+
+# %% [markdown]
+# Then we can retrieve our final dataset from the active learning steps
+
+# %%
+dataset = result.try_get_final_dataset()
+query_points = dataset.query_points.numpy()
+observations = dataset.observations.numpy()
+
+# %% [markdown]
+# Finally, we can check the performance of our `PredictiveVariance` active learning acquisition function by plotting the predictive variance landscape of our model. We can see how it samples on regions with high variance value.
```

Comment:
Finally, we can check the performance of our `PredictiveVariance` active learning acquisition function by plotting the predictive variance landscape of our model. We can see how it samples regions for which our model is highly uncertain.